### PR TITLE
Button: v.2. Added support for href and segmented Flow type

### PR DESCRIPTION
**Pull request by AlbertCarreras**
Changes documented here: [TDD](https://paper.dropbox.com/doc/Gestalt-Button-V.2--A6cM36Ezpyk_HZcDKcVTs74HAg-gNyH1dOoWYwvPfqm3Kgme)

- Added support to href to Button.
- Segmented Flow for each Button type to make Flow Type stricter in Button.
- Upgraded Buttons Docs.
- Added test.
- Remove `name` from `Button`

### CODE REVIEW

File: docs/src/Button.doc.js [hunk 1]
```diff
@@ -127,7 +127,7 @@ card(
   <Example
     name="Example"
     defaultCode={`
-<Button text="Medium sized button" inline />
+<Button href="http://www.pinterest.com" text="Medium sized button" target='blank' type="link" inline />
```
Comment (christianvuerings) [inline docs/src/Button.doc.js:130]: @AlbertCarreras let's not make this the default example and add a new one instead.

File: packages/gestalt/src/Button.js [hunk 2]
```diff
@@ -28,54 +29,65 @@ const DEFAULT_TEXT_COLORS = {
...
+  name?: string,
+  type: 'submit',
+|};
+
+type LinkButtonType = {|
+  ...BaseButton,
+  href: string,
+  rel?: 'none' | 'nofollow',
+  target?: null | 'self' | 'blank',
+  type: 'link',
```
Comment (christianvuerings) [inline packages/gestalt/src/Button.js:68]: @AlbertCarreras just so I understand. Does this mean developers would have to add `type="link"` each time they add the `href` tag?

File: packages/gestalt/src/Button.js [hunk 3]
```diff
@@ -143,6 +155,85 @@ const ButtonWithForwardRef: React$AbstractComponent<
...
+    const { href, rel, target } = props;
+
+    return (
+      <Link
+        hoverStyle="none"
+        href={href}
+        ref={innerRef}
+        rel={rel}
+        target={target}
+        inline
```
Comment (christianvuerings) [inline packages/gestalt/src/Button.js:168]: @AlbertCarreras curious to know what happens for non-inline buttons. Should this be conditional based on the passed in props?

File: packages/gestalt/src/Button.js [hunk 4]
```diff
@@ -143,6 +155,85 @@ const ButtonWithForwardRef: React$AbstractComponent<
...
+        rounding="pill"
+      >
+        <div
+          style={{ alignItems: 'center', display: 'flex' }}
```
Comment (christianvuerings) [inline packages/gestalt/src/Button.js:172]: @AlbertCarreras let's add it to a CSS file instead of using inline styles. (perf)

File: packages/gestalt/src/Button.js [hunk 5]
```diff
@@ -143,6 +155,85 @@ const ButtonWithForwardRef: React$AbstractComponent<
...
+        onMouseUp={handleMouseUp}
+        onTouchCancel={handleTouchCancel}
+        onTouchEnd={handleTouchEnd}
+        onTouchMove={handleTouchMove}
+        onTouchStart={handleTouchStart}
+        ref={innerRef}
+        type={type}
+        {...(compressStyle ? { style: compressStyle } : {})}
+      >
+        {iconEnd ? iconEndComponent(iconEnd) : buttonText}
```
Comment (christianvuerings) [inline packages/gestalt/src/Button.js:224]: @AlbertCarreras does this mean we can not have `iconEnd` on `Link`? I think that's a pretty big use case (e.g. external links that look like a button)

File: packages/gestalt/src/Button.js [hunk 6]
```diff
@@ -28,54 +29,65 @@ const DEFAULT_TEXT_COLORS = {
...
     disabled = false,
-    iconEnd,
     inline = false,
-    name,
     onClick,
     selected = false,
     size = 'md',
     text,
     textColor: textColorProp,
-    type = 'button',
```
Comment (christianvuerings) [inline packages/gestalt/src/Button.js:77]: @AlbertCarreras could we leave `type = 'button'` as the default? Otherwise the default becomes "submit" and this PR becomes a breaking change which will most likely cause bugs in Pinterest's codebase.

image

See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-type

File: docs/src/Button.doc.js (hunk 1 above)
Comment (AlbertCarreras) [reply docs/src/Button.doc.js:130]: Oh I know, I did this because it's how I'm testing while developing

File: packages/gestalt/src/Button.js (hunk 2 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Button.js:68]: It means that. We could just use href to define the type we return but I think for communication clarity it's easier to explain that we have 3 types each of them requiring specific props. `submit` requires name, `link` requires href, etc. 

My thoughts, we could use just href though but href excludes type submit/button then and it has to be described rather than being explicit with type link.

File: packages/gestalt/src/Button.js (hunk 3 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Button.js:168]: WIP. The finishings were not done.

File: packages/gestalt/src/Button.js (hunk 4 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Button.js:172]: Ok. Makes sense. It was easy for me to see if I was correcting it.

File: packages/gestalt/src/Button.js (hunk 5 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Button.js:224]: Agree. While coding it got in the mindset that icons were not needed for links. Again, WIP, finishings were not done.

File: packages/gestalt/src/Button.js (hunk 6 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Button.js:77]: I probably missed this. The intention was to leave the button as default. The draft was more of the first exploratory implementation. But thanks for catching that.

File: packages/gestalt/src/Button.js (hunk 4 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Button.js:172]: DONE

File: packages/gestalt/src/Button.js (hunk 3 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Button.js:168]: DONE

File: packages/gestalt/src/Button.js (hunk 5 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Button.js:224]: DONE

File: packages/gestalt/src/Button.js (hunk 6 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Button.js:77]: See line 231
image

Comment (AlbertCarreras) [reply packages/gestalt/src/Button.js:77]: DONE

File: docs/src/Button.doc.js [hunk 7]
```diff
@@ -9,281 +9,287 @@ import PageHeader from './components/PageHeader.js';
 const cards: Array<Node> = [];
 const card = c => cards.push(c);
 
+const buildDescription = (lines: Array<string>): Node => (
```
Comment (christianvuerings) [inline docs/src/Button.doc.js:12]: @AlbertCarreras Instead of adding this to `Button.doc.js`, could we move it to `PropTable`? Basically if `description` is an array of strings, execute this logic.

File: docs/src/Button.doc.js [hunk 8]
```diff
@@ -9,281 +9,287 @@ import PageHeader from './components/PageHeader.js';
...
+      {
+        name: 'accessibilityLabel',
+        type: 'string',
+        required: false,
+        defaultValue: null,
+        description: buildDescription([
+          'Supply a short, descriptive label for screen-readers to replace button texts that do not provide sufficient context about the button component behavior. Texts like `Click Here,` `Follow,` or `Read More` can be confusing when a screen reader reads them out of context. In those cases, we must pass an alternative text to replace the button text.',
+          'Use IconButton if you only need a button with an icon and no text.',
+          'Optional with button-type and submit-type buttons.',
+          'Accessibility: It populates aria-label.  Screen readers read the `accessibilityLabel` prop, if present, instead of `text`.',
```
Comment (christianvuerings) [inline docs/src/Button.doc.js:41]: @AlbertCarreras could you replace the double spaces? `.  ` => `. `

File: docs/src/Button.doc.js [hunk 9]
```diff
@@ -9,281 +9,287 @@ import PageHeader from './components/PageHeader.js';
...
+        ]),
+        href: 'accessibilityI',
```
Comment (christianvuerings) [inline docs/src/Button.doc.js:43]: @AlbertCarreras instead, can we be more descriptive and link to `accessibilityLabel` instead?

Same for `accessibilityII`, which could be something like `accessibility` as the ID and `Accessibility: controls & expanded` as the title.

File: docs/src/Button.doc.js [hunk 10]
```diff
@@ -9,281 +9,287 @@ import PageHeader from './components/PageHeader.js';
...
+          'Indicate that a button component controls the appearance of menus and expose whether they are currently opened or closed.',
+          'Optional with button-type buttons.',
+          'Accessibility: It populates aria-haspopup.',
+        ]),
+        href: 'accessibilityII',
       },
       {
         name: 'color',
-        type: `"blue" | "gray" | "red" | "transparent" | "white"`,
+        type: `gray' | 'red' | 'blue' | 'transparent' | 'white'`,
```
Comment (christianvuerings) [inline docs/src/Button.doc.js:83]: @AlbertCarreras I think there is a `'` missing before `gray`

File: docs/src/Button.doc.js [hunk 11]
```diff
@@ -9,281 +9,287 @@ import PageHeader from './components/PageHeader.js';
...
+          'Add a Gestalt icon to be displayed after the button text. Sometimes an icon can help clarify the usage of a button. Menus are a common use case.',
+          'Accessibility: Icons on buttons are not accessible for screen readers.',
+          'Use IconButton if you only need buttons with icons and no text.',
+        ]),
         href: 'iconEnd',
       },
       {
         name: 'inline',
         type: 'boolean',
+        required: false,
```
Comment (christianvuerings) [inline docs/src/Button.doc.js:113]: @AlbertCarreras curious why we added `required: false`, is that not the default?

File: docs/src/Button.doc.js [hunk 12]
```diff
@@ -9,281 +9,287 @@ import PageHeader from './components/PageHeader.js';
...
+          'Display a button in different sizes. Size only changes top & bottom padding modifying the component height: sm (32px), md (40px), lg (48px).',
+        href: 'height',
       },
       {
         name: 'text',
         type: 'string',
         required: true,
+        defaultValue: null,
+        description: buildDescription([
+          'Text to render inside the button to convey the function and purpose of the button. The button text has a fix size.',
```
Comment (christianvuerings) [inline docs/src/Button.doc.js:145]: `fix size` => `fixed size`

File: docs/src/Button.doc.js [hunk 13]
```diff
@@ -9,281 +9,287 @@ import PageHeader from './components/PageHeader.js';
...
+        defaultValue: null,
+        description: buildDescription([
+          'Primary colors to apply to the text inside the Button.',
+          'Text colors are automatically paired with the defined button `color`.',
+          '`textColor` should only be used when using a white background color; it may become necessary to manually override the text color to match that of the parent`s background. Do not override the text color for any button besides those with white background as it breaks design standards.',
+        ]),
         href: 'textColor',
       },
       {
         name: 'type',
```
Comment (christianvuerings) [inline docs/src/Button.doc.js:163]: @AlbertCarreras thought a bit more about it over the weekend and I don't think we should switch on `type`. Mostly because we'll want a consistent API with `TapArea`.

What do you think about keeping `type` as is, with `button | submit` and introducing `role` with `button | link`?

```jsx
<TapArea role="link" href="https://pinterest.com" />
<Button role="link" href="https://pinterest.com" />
```

File: docs/src/Button.doc.js [hunk 14]
```diff
@@ -9,281 +9,287 @@ import PageHeader from './components/PageHeader.js';
...
-        type: `"submit" | "button"`,
+        type: `'submit' | 'button' | 'link'`,
+        required: false,
         defaultValue: 'button',
-        href: 'types',
+        description: buildDescription([
+          'Select a button variant:',
+          '-`button`: Use to render simple push buttons with no default behavior and control custom functionality inside the `onClick` callback. The button is rendered as an button or  `<button> tag`.',
```
Comment (christianvuerings) [inline docs/src/Button.doc.js:169]: Replace `an` with `a` and remove double spaces after `or`

```
The button is rendered as an button or  `<button> tag`.
```

```
The button is rendered as a button or `<button> tag`.
```

File: docs/src/Button.doc.js [hunk 15]
```diff
@@ -9,281 +9,287 @@ import PageHeader from './components/PageHeader.js';
...
-        type: "React.Ref<'button'>",
-        description: 'Forward the ref to the underlying button element',
-        href: 'refExample',
+        required: false,
+        defaultValue: null,
+        description: buildDescription([
+          'Callback fired when a button component is clicked (pressed and released) with a mouse or keyboard.',
+          'Required with button-type buttons.',
+        ]),
+        href: 'selected',
```
Comment (christianvuerings) [inline docs/src/Button.doc.js:128]: @AlbertCarreras should this `href` move to the `selected` prop?

File: docs/src/Button.doc.js [hunk 16]
```diff
@@ -9,281 +9,287 @@ import PageHeader from './components/PageHeader.js';
...
+        ]),
+        href: '',
+      },
+      {
+        name: 'href',
+        type: 'string',
+        required: false,
+        defaultValue: null,
+        description: buildDescription([
+          'Specify a redirect url.',
```
Comment (christianvuerings) [inline docs/src/Button.doc.js:193]: @AlbertCarreras not really a redirect, let's rephrase this as `Specify a link URL.`

File: docs/src/Button.doc.js [hunk 17]
```diff
@@ -9,281 +9,287 @@ import PageHeader from './components/PageHeader.js';
...
-    <Box display="flex" direction="row" column={6} paddingX={2}>
-      <Button color="transparent" text="Later" />
-    </Box>
-    <Box column={6} paddingX={2}>
-      <Button color="white" text="Learn more" />
-    </Box>
-  </Box>
-</Box>
+<Row gap={1}>
+    <Tooltip text="button-type button">
```
Comment (christianvuerings) [inline docs/src/Button.doc.js:247]: @AlbertCarreras could we fix the indentation to be 2 spaces?

File: packages/gestalt/src/Button.flowtest.js [hunk 18]
```diff
@@ -4,8 +4,10 @@ import Button from './Button.js';
 
 const Valid = <Button text="Next" />;
 
-// $FlowExpectedError[prop-missing]
```
Comment (christianvuerings) [inline packages/gestalt/src/Button.flowtest.js:7]: @AlbertCarreras could we add a couple more tests of buttons that are correct and incorrect to test the flow types?

File: packages/gestalt/src/Button.js [hunk 19]
```diff
@@ -28,54 +29,67 @@ const DEFAULT_TEXT_COLORS = {
...
   textColor?: 'white' | 'darkGray' | 'blue' | 'red',
-  type?: 'submit' | 'button',
 |};
 
+type ButtonType = {|
+  ...BaseButton,
+  accessibilityControls?: string,
+  accessibilityExpanded?: boolean,
+  accessibilityHaspopup?: boolean,
+  accessibilityLabel?: string,
```
Comment (christianvuerings) [inline packages/gestalt/src/Button.js:54]: @AlbertCarreras `accessibilityLabel?: string` should be moved to `BaseButton`. It's totally fine to set a label on a `Link` but not required.

We can also add it to `Link` if that prop isn't on that component yet.

File: packages/gestalt/src/Button.js [hunk 20]
```diff
@@ -142,7 +157,89 @@ const ButtonWithForwardRef: React$AbstractComponent<
...
+
   /* eslint-disable react/button-has-type */
+  if (props.type === 'link') {
+    const { href, rel, target } = props;
+
+    return (
+      <Link
+        hoverStyle="none"
+        href={href}
+        onClick={({ event }) => {
```
Comment (christianvuerings) [inline packages/gestalt/src/Button.js:190]: @AlbertCarreras could we extract this `onClick` + the one below into 1 `handleClick`?

```jsx
function handleClick(event) {
  if (onClick) {
    onClick({event})
  }
}
///
<Link onClick={handleClick} />
```

File: packages/gestalt/src/Button.js [hunk 21]
```diff
@@ -142,7 +157,89 @@ const ButtonWithForwardRef: React$AbstractComponent<
...
+          if (onClick) {
+            onClick({ event });
+          }
+        }}
+        ref={innerRef}
+        rel={rel}
+        target={target}
+        inline={inline}
+        rounding="pill"
+        tapStyle="compress"
```
Comment (christianvuerings) [inline packages/gestalt/src/Button.js:200]: @AlbertCarreras doesn't this mean we have 2 compress styles?
1. Within `classes` on the wrapping `div`
2. Within `Link`

If so, we probably want to keep the one on `Link` and make the one on `classes` conditional.

```
[touchableStyles.tapCompress]: props.type !== 'link' && !disabled && isTapping
```

File: packages/gestalt/src/Button.js [hunk 22]
```diff
@@ -142,7 +157,89 @@ const ButtonWithForwardRef: React$AbstractComponent<
...
+            icon={iconEnd}
+            size={SIZE_NAME_TO_PIXEL[size]}
+          />
+        </Box>
+      </Box>
+    );
+  };
+
   /* eslint-disable react/button-has-type */
+  if (props.type === 'link') {
```
Comment (christianvuerings) [inline packages/gestalt/src/Button.js:183]: @AlbertCarreras instead of `props.type`, can we compare with `type` and destructure above?

File: packages/gestalt/src/__snapshots__/Toast.test.js.snap [hunk 23]
```diff
@@ -106,7 +106,7 @@ exports[`<Toast /> Text + Image + Button 1`] = `
           onTouchEnd={[Function]}
           onTouchMove={[Function]}
           onTouchStart={[Function]}
-          type="button"
+          type="Button"
```
Comment (christianvuerings) [inline packages/gestalt/src/__snapshots__/Toast.test.js.snap:109]: @AlbertCarreras this is a bug - it should be `button` lowercase instead.

**Comment by christianvuerings** [comment]
@AlbertCarreras seeing a double scrollbar on the proptable now. Could you investigate?

FYI we do need the `overflow` auto so the table becomes scrollable with a horizontal scrollbar.

image

File: docs/src/Button.doc.js [hunk 24]
```diff
@@ -295,8 +301,7 @@ function MenuButtonExample() {
         <Button
           accessibilityExpanded={!!open}
           accessibilityHaspopup
-          color="white"
-          iconEnd="arrow-down"
+          iconEnd={open ? "arrow-down" : "arrow-forward"}
```
Comment (christianvuerings) [inline docs/src/Button.doc.js:304]: @AlbertCarreras could we revert the dropdown icon change? Instead, we should toggle the `selected` state when the dropdown is open (please verify with design)

File: docs/src/Button.doc.js [hunk 25]
```diff
@@ -428,42 +449,58 @@ function A11yExDisclosure() {
...
-function ButtonFlyoutExample() {
-  const [open, setOpen] = React.useState(false);
-  const [checked, setChecked] = React.useState(false);
+function AccessibilityExample() {
+  const [isFollowing, setIsFollowing] = React.useState(false);
 
-  const anchorRef = React.useRef(null);
+  return (
+    <Button
+      accessibilityLabel={isFollowing ? "Following Alberto on Pinterest" : "Follow Alberto on Pinterestlow"}
```
Comment (christianvuerings) [inline docs/src/Button.doc.js:460]: @AlbertCarreras could we make this
```jsx
accessibilityLabel={isFollowing ? "Unfollow Alberto on Pinterest" : "Follow Alberto on Pinterest"}
text={isFollowing ? "Unfollow" : "Follow"}
```

File: docs/src/Button.doc.js [hunk 26]
```diff
@@ -428,42 +449,58 @@ function A11yExDisclosure() {
...
       <Button
+        accessibilityControls={COLLAPSIBLE_MENU}
+        accessibilityExpanded={isOpen}
+        accessibilityHasPopup="true"
         inline
-        color="red"
-        onClick={() => setOpen(true)}
-        ref={anchorRef}
-        text="Submit"
+        iconEnd={isOpen ? "arrow-down" : "arrow-forward"}
```
Comment (christianvuerings) [inline docs/src/Button.doc.js:487]: @AlbertCarreras same here - let's check with design whether we actually want to toggle the icon

File: docs/src/Button.doc.js (hunk 7 above)
Comment (AlbertCarreras) [reply docs/src/Button.doc.js:12]: That makes more sense. Thanks

File: docs/src/Button.doc.js (hunk 10 above)
Comment (AlbertCarreras) [reply docs/src/Button.doc.js:83]: You think... no... there isnt!

File: docs/src/Button.doc.js (hunk 7 above)
Comment (AlbertCarreras) [reply docs/src/Button.doc.js:12]: DONE

File: docs/src/Button.doc.js [hunk 27]
```diff
@@ -12,217 +12,390 @@ const card = c => cards.push(c);
...
     defaultCode={`
-<Button text="Medium sized button" inline />
+<Row gap={1}>
+  <Tooltip text="default button: button-role + button-type button">
+    <Button onClick={() => {}} text="Clear search history" inline />
+  </Tooltip>
+  <Tooltip text="form button variant: button-role + submit-type button">
+    <Button type="submit" name='satisfaction-questionaire' text="Submit your response" inline />
+  </Tooltip>
+  <Tooltip text="link button variant: link-role button + button-type">
```
Comment (christianvuerings) [inline docs/src/Button.doc.js:253]: Could we simplify the tooltip text? Just show `Link button` / `Submit button` / `Default button` in the tooltip. Also don't think `link-role button + button-type` is what we want

File: docs/src/Button.doc.js [hunk 28]
```diff
@@ -12,217 +12,390 @@ const card = c => cards.push(c);
...
         name: 'size',
-        type: `"sm" | "md" | "lg"`,
+        type: `'sm' | 'md' | 'lg'`,
+        required: false,
         defaultValue: 'md',
-        description: 'sm: 32px, md: 40px, lg: 48px',
-        href: 'combinations',
+        description:
+          'Display a button in different sizes. Size only changes top & bottom padding modifying the component height: sm (32px), md (40px), lg (48px).',
+        href: 'height',
```
Comment (christianvuerings) [inline docs/src/Button.doc.js:126]: @AlbertCarreras could we rename `height` to `sizes`. It not only updates the top/bottom padding but also left/right.

File: packages/gestalt/src/Button.js [hunk 29]
```diff
@@ -122,9 +139,11 @@ const ButtonWithForwardRef: React$AbstractComponent<
       [styles.selected]: !disabled && selected,
       [styles.disabled]: disabled,
       [styles.enabled]: !disabled,
```
Comment (christianvuerings) [inline packages/gestalt/src/Button.js:141]: @AlbertCarreras looks like a disabled `<Button href="" disabled />` does not work as expected:

* The button (link) is still clickable
* It still opens the link

image

File: packages/gestalt/src/Button.jsdom.test.js [hunk 30]
```diff
@@ -13,10 +13,21 @@ describe('Button', () => {
     expect(mockOnClick).toHaveBeenCalled();
   });
 
-  it('forwards a ref to the innermost button element', () => {
-    const ref = React.createRef();
+  it('forwards a ref to the innermost button element with type button', () => {
```
Comment (christianvuerings) [inline packages/gestalt/src/Button.jsdom.test.js:16]: Could we add a test for a disabled Button with `href`?

File: docs/src/Button.doc.js (hunk 24 above)
Comment (AlbertCarreras) [reply docs/src/Button.doc.js]: DONE.

File: docs/src/Button.doc.js (hunk 26 above)
Comment (AlbertCarreras) [reply docs/src/Button.doc.js]: DONE

File: docs/src/Button.doc.js (hunk 27 above)
Comment (AlbertCarreras) [reply docs/src/Button.doc.js:253]: DONE

File: packages/gestalt/src/Button.js (hunk 22 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Button.js]: can't, it triggers Flow errors

File: docs/src/Button.doc.js (hunk 13 above)
Comment (AlbertCarreras) [reply docs/src/Button.doc.js:152]: DONE

File: docs/src/Button.doc.js (hunk 28 above)
Comment (AlbertCarreras) [reply docs/src/Button.doc.js:126]: DONE Yeah, didn't notice lg also changes left/right padding

File: packages/gestalt/src/Button.js (hunk 29 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Button.js:141]: I added _disable prop to Link so I can do `href={ _disable ? undefined : href}`

We can use this approach for id, role and, accessibilitySelected so we know they are private props and therefore not documented

File: packages/gestalt/src/Button.jsdom.test.js (hunk 30 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Button.jsdom.test.js]: DONE

File: packages/gestalt/src/Button.js (hunk 29 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Button.js:141]: DONE

**Comment by AlbertCarreras** [comment]
> @AlbertCarreras seeing a double scrollbar on the proptable now. Could you investigate?
> 
> FYI we do need the `overflow` auto so the table becomes scrollable with a horizontal scrollbar.
> 
> image
> 
> Looks like it's caused by `buildDescription`

DONE

File: docs/src/components/PropTable.js [hunk 31]
```diff
@@ -98,7 +98,10 @@ export default function PropTable({
 
   return (
     <Card id="Props" name="Props" showHeading={showHeading}>
-      <Box overflow="auto">
+      <Box
+        overflow="auto"
+        dangerouslySetInlineStyle={{ __style: { overflowY: 'hidden' } }}
```
Comment (christianvuerings) [inline docs/src/components/PropTable.js:103]: @AlbertCarreras 🤭  - not for this diff but I've been thinking we should add `overflow="autoX"` and `overflow="autoY"` to `Box` so we don't need to do this hack.

File: packages/gestalt/src/Button.js [hunk 32]
```diff
@@ -142,7 +161,108 @@ const ButtonWithForwardRef: React$AbstractComponent<
...
+  }
+
   /* eslint-disable react/button-has-type */
+  if (props.role === 'link') {
+    const { href, rel, target } = props;
+
+    return (
+      <Link
+        accessibilityLabel={accessibilityLabel}
+        _disabled={disabled}
```
Comment (christianvuerings) [inline packages/gestalt/src/Button.js:215]: @AlbertCarreras Let's not make this a pattern and add `_` props for "private" props. People can still use them.

Instead, let's add `disabled` to `Link`

File: docs/src/Button.doc.js [hunk 33]
```diff
@@ -12,217 +12,410 @@ const card = c => cards.push(c);
...
       },
       {
         name: 'inline',
         type: 'boolean',
+        required: false,
         defaultValue: false,
-        href: 'widths',
-      },
-      {
-        name: 'name',
```
Comment (christianvuerings) [inline docs/src/Button.doc.js:84]: @AlbertCarreras let's make it explicit that we removed `name` from `Button` by adding a codemod which removes this prop. Even though we might not use it in the Pinterest codebase, we should still be due diligent.

**Comment by christianvuerings** [comment]
@AlbertCarreras feel free to self merge after:
* Renaming `_disabled` on `Link` to `disabled`
* Adding a codemod to remove `name` from all `<Button />`s

File: docs/src/Button.doc.js (hunk 33 above)
Comment (AlbertCarreras) [reply docs/src/Button.doc.js:84]: @cvuerings, I didn't remove it. Just undocumented.

Comment (AlbertCarreras) [reply docs/src/Button.doc.js:84]: We still have a few instances on Pinboard that we could rename to `id`

File: packages/gestalt/src/Button.js (hunk 32 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Button.js:215]: DONE
